packages/notte-core/src/notte_core/ast.py

```
@@ -840,17 +917,18 @@ def run_script(self, code_string: str, variables: dict[str, Any] | None = None,
-
-                # Call the run function
-                run_ft = execution_globals.get("run")
-                if run_ft is None or not callable(run_ft):
-                    raise MissingRunFunctionError("Python script must contain a 'run' function")
-                if callable(run_ft):
-                    return run_ft(**variables) if variables else run_ft()  # pyright: ignore [reportUnknownVariableType]
-
-                return execution_globals
+                with deny_subprocess_creation():
+                    # Execute the script in regular Python
+                    exec(parsed_info.code, execution_globals)
```

Ruff (0.16.2)

[error] Use of `exec` detected

(S102)